oxide-auth/src/primitives/generator.rs

```diff
@@ -49,22 +49,22 @@ pub trait TagGrant {
 /// Each byte is chosen randomly from the basic `rand::thread_rng`. This generator will always
 /// succeed.
 pub struct RandomGenerator {
-    random: SystemRandom,
+    random: OsRng,
     len: usize
 }
 
 impl RandomGenerator {
     /// Generates tokens with a specific byte length.
     pub fn new(length: usize) -> RandomGenerator {
         RandomGenerator {
-            random: SystemRandom::new(),
+            random: OsRng {},
             len: length
         }
     }
 
-    fn generate(&self) -> String {
+    fn generate(&mut self) -> String {
```

Comment:
That's a surprising change in signature and leads to a bunch of unwarranted removals of impls down the road. `OsRng` is `Copy` and there should be no need for that.

Reply:
Totally right, will `Clone()`.